### DISCUSSION TITLE
Argo Workflows as a FaaS tool?

### DISCUSSION THREAD

**Discussion by Brian-V** [argoproj/argo-workflows#11730] [General]
One of my colleagues shared the opinion that Argo Workflows could be used as a Function as a Service capability. I’m more familiar with the Argo CD tool, not workflows. But I haven’t found anywhere through Google where folks are talking about Argo Workflows for FaaS. I’d love to hear opinions from this community. Thanks!

**Comment by agilgur5** [comment]
In AWS terminology, I would say Workflows is more similar to Step Functions and Fargate than pure FaaS like Lambda. The main selling point is to be able to orchestrate complex DAGs natively in Kubernetes.
Step Functions allow some logic between Lambdas and Fargate runs "serverless" containers (as opposed to functions). Workflows does not provide a runtime like Lambda and instead runs containers. Although there are some built-in templates like [`http`](https://argoproj.github.io/argo-workflows/http-template/) to provide common functionality.

If you use Python, [Hera](https://github.com/argoproj-labs/hera/) is an SDK that abstracts some low-level details of Workflows so that you can write mostly pure Python functions and deploy those. That makes it a bit more similar to Lambda with some sort of runtime.

[Argo Events](https://github.com/argoproj/argo-events) can also be used in conjunction to Workflows to add various sensors/triggers, similar to EventBridge.

[Knative](https://github.com/knative/docs) is a more generalized FaaS for Kubernetes tool.

And also, since this is k8s, you still need to scale your cluster independently compared to "serverless".

To summarize, I would say yesn't, or kind of. Workflows serves a different purpose, but may or may not fit your needs for a FaaS tool. It depends on your requirements.

**Reply by Brian-V** [reply]
Thank you! Thanks so much for the thoughtful response. I think I get it.
